### PR TITLE
Persist name local storage

### PR DESCRIPTION
**Pull request by SupriyaAdep**
Issue description: https://app.clickup.com/t/51e5kp
Store the display name in localstorage and only enable the join room button when display name exist. (name is mandatory)
Tested on: 
a) Web
b) iOS
c) Electron macOS

### CODE REVIEW

File: template/src/components/Precall.tsx [hunk 1]
```diff
@@ -154,17 +154,18 @@ const Precall = (props: any) => {
...
+                  placeholder={
+                    queryComplete ? 'Display name*' : 'Getting name...'
+                  }
+                  editable={queryComplete}
                 />
                 <View style={{height: 20}} />
                 <PrimaryButton
                   onPress={() => setCallActive(true)}
-                  disabled={!queryComplete}
+                  disabled={!queryComplete || username.trim() == ''}
```
Comment (nitte93) [inline template/src/components/Precall.tsx:168]: use ===

File: template/src/components/Precall.tsx [hunk 2]
```diff
@@ -94,17 +89,18 @@ const Precall = (props: any) => {
...
+                placeholder={
+                  queryComplete ? 'Display name*' : 'Getting name...'
+                }
+                editable={queryComplete}
               />
               <View style={style.margin5Btm} />
               <PrimaryButton
                 onPress={() => setCallActive(true)}
-                disabled={!queryComplete}
+                disabled={!queryComplete || username.trim() == ''}
```
Comment (nitte93) [inline template/src/components/Precall.tsx:103]: use ===

File: template/src/components/Precall.native.tsx [hunk 3]
```diff
@@ -95,10 +99,14 @@ const Precall = (props: any) => {
...
         {/* <Text style={style.buttonText}> */}
-          {/* {queryComplete ? 'Join Room' : 'Loading...'} */}
+        {/* {queryComplete ? 'Join Room' : 'Loading...'} */}
         {/* </Text> */}
-      {/* </TouchableOpacity> */}
-      <PrimaryButton text={"Join Room"} disabled={!queryComplete} onPress={() => setCallActive(true)} />
+        {/* </TouchableOpacity> */}
+        <PrimaryButton
+          text={'Join Room'}
+          disabled={!queryComplete || username.trim() == ''}
```
Comment (nitte93) [inline template/src/components/Precall.native.tsx:107]: ===

File: template/src/pages/VideoCall.tsx [hunk 4]
```diff
@@ -212,8 +218,10 @@ enum RnEncryptionEnum {
 }
 
 const VideoCall: React.FC = () => {
-  const {store} = useContext(StorageContext);
-  const [username, setUsername] = useState('Getting name...');
+  const {store, setStore} = useContext(StorageContext);
+  const [username, setUsername] = useState(
```
Comment (nitte93) [inline template/src/pages/VideoCall.tsx:222]: use lazily initialise useState for conditional or anything other than a constant value.
const [username, setUsername] = useState( () =>   store?.displayName ? store.displayName : '',);

**Review by nitte93 — CHANGES_REQUESTED** [review]

File: template/src/components/Precall.tsx (hunk 2 above)
Comment (SupriyaAdep) [reply template/src/components/Precall.tsx]: updated

File: template/src/components/Precall.native.tsx (hunk 3 above)
Comment (SupriyaAdep) [reply template/src/components/Precall.native.tsx]: updated

File: template/src/pages/VideoCall.tsx (hunk 4 above)
Comment (SupriyaAdep) [reply template/src/pages/VideoCall.tsx]: updated

**Review by nitte93 — APPROVED** [review]

**Comment by nitte93** [comment]
LGTM 👍

**Pull request merged**